cgi-bin/LJ/Sysban.pm

```diff
@@ -198,6 +198,42 @@ sub sysban_check {
             }, undef, $wh, $vl);
     };
 
+    # helper for iteratively checking all subdomains of a hostname
+    my $check_all_subdomains = sub {
```

Comment:
Since we're going to hit the database and load every matching element, and we're doing this just for `email_domain` types, then we should instead of doing all this magic just:

```
SELECT value FROM sysban
WHERE status = 'active'
     AND what = ?
     AND (value = ? OR ? LIKE CONCAT("%.", value))
     AND NOW() > bandate
     AND (NOW() < banuntil
              OR banuntil = 0
              OR banuntil IS NULL)
```

And pass `('email_domain', $domain, $domain)` (yes twice, to allow exact matches, and to prevent "foobar.com" matching against "bar.com").

This gets rid of nearly all of the Perl code and makes the database do it for us. Thoughts?